---
scene_setup/BlockArray arrangement